src/operators/suffix.ts

```diff
@@ -139,6 +139,23 @@ export class SuffixOperator implements Operator {
     }
   }
 
+  /**
+   * Check to see if a prop already contains a suffix
+   * @param prop The prop to check
+   * @return If the prop has a suffix on it already
+   */
+  static isAlreadySuffixed(prop: string) : boolean {
+    if (prop === undefined || prop === null) {
```

Comment:
What do you think about a quick test if the char at index N has an `_`?  Since we know the suffixes, we could perform a direct access test to see if a suffix looks to be present.  If so, then we do the subsequent test (the loop) to verify if a valid suffix is present.

Reply:
We could do that.  We'd have to check for length - 3 and length - 4 for the various sizes.  If someone were to add a longer suffix, we'd have to update the code.  I tried to make it so it was more generic, just working off the enum.

Reply:
Agree that new suffixes would require a change.  I'm thinking that a property with an existing suffix is low probability since this data is coming from the data layer and would likely never have a FullStory-specific suffix.  So the quick test would efficiently check before doing a more exhaustive check of all possible suffixes.

Reply:
I can put that in if folks want.  With library code that can live a long time, I usually err on the side of not putting in potential stumbling blocks, and hard coding a position looking for suffixes that could change if someone adds a suffix is against my nature.  Not a hill for me to die on, so if you folks want it, I can put that in.  endsWith is not a super expensive operator

Reply:
Also in looking at our suffixes, we have some that are 3, 4 and 5 in length.  To do a quick test to see if they have the _ in the right position, I'd have to look for it at all three of those positions.  I could look for the presence of a _ character, but not sure that is any faster than endsWith.  In the examples posted, it looks like the split the value by the _ character, but I'm not sure that is any faster.  If something isn't much faster, I'd vote for readability.  In this case, it is a suffix operator, which means a string endsWith.

Reply:
New suffixes shouldn't cause an issue, and suffix length shouldn't be a problem either if you snag a single substring based on the last index of `_` and check that against an object with keys containing known suffix. This is what our Segment cloud mode destination and our FS client APIs do (e.g. `FS.setUserVars`). https://github.com/fullstorydev/fullstory-data-layer-observer/pull/208#discussion_r1054530767